src/commands/createContainerApp/ContainerAppCreateStep.ts

```diff
@@ -64,7 +64,7 @@ export class ContainerAppCreateStep extends AzureWizardExecuteStep<IContainerApp
         ext.outputChannel.appendLog(creatingSwa);
 
         context.image ||= `${getLoginServer(context)}/${context.repositoryName}:${context.tag}`;
-        const name = context.image.substring(context.image.lastIndexOf('/') + 1).replace(':', '-');
+        const name = context.image.substring(context.image.lastIndexOf('/') + 1).replace(/[^0-9a-zA-Z-]/g, '-');
```

Comment:
I think this should be a function, since this regex is repeated twice, and they should always be kept in sync (right?).

Reply:
Maybe something like `getContainerNameForImage(image: string): string`?

Reply:
Yeah, good point.  I thought the same thing, but figured I'd just overhaul this entire extension whenever we do the unified work.  This one has become a mess because of some last minute pivots in design 😬

Reply:
If it's helpful, we have some similar [stuff](https://github.com/microsoft/vscode-docker/blob/main/src/utils/getValidImageName.ts) in the Docker extension that has been stable for years.